[docs] appearance, formatting and wording tweaks/fixes (#29567)
expo/expo: docs/pages/ui-programming/react-native-styling-buttons.mdx

import ImageSpotlight from '~/components/plugins/ImageSpotlight';
import { SnackInline } from '~/ui/components/Snippet';

> React Native exports a [`<Button>`](https://reactnative.dev/docs/button) component that exposes the native button element for Android, iOS, and the web. The `<Button>` component accepts `title` and `onPress` props, but it does not accept a `style` prop, which makes it hard to customize the style.

Reviewer: ```suggestion
React Native exports a [`<Button>`](https://reactnative.dev/docs/button) component that exposes the native button element for Android, iOS, and the web. This component accepts `title` and `onPress` props. It does not accept a `style` prop and doesn't support custom styling.
```

Let's improve the verbiage here.
